modules/util/portal-tools-rest-builder/src/main/resources/com/liferay/portal/tools/rest/builder/dependencies/base_resource_impl.ftl

```diff
@@ -975,69 +979,82 @@ public abstract class Base${schemaName}ResourceImpl
 		}
 	}
 
-	private Collection<Permission> _getPermissions(long companyId, List<ResourceAction> resourceActions, long resourceId, String resourceName, String[] roleNames) throws Exception {
+	private Collection<Permission> _getPermissions( long companyId, List<ResourceAction> resourceActions,
+		long resourceId, String resourceName, String[] roleNames)
+			throws Exception {
+
 		_checkResources(companyId, resourceId, resourceName);
 
 		Map<String, Permission> permissions = new LinkedHashMap<>();
 
-		List<ResourcePermission> resourcePermissions = new ArrayList<>();
+		List<String> actionIds = new ArrayList<>();
 
-		try {
-			Method method = ResourcePermissionLocalService.class.getMethod("getResourcePermissions", String.class);
+		Set<ResourcePermission> resourcePermissionSet = _getResourcePermissionSet(
+			companyId, resourceId, resourceName);
 
-			resourcePermissions = (List<ResourcePermission>)method.invoke(resourcePermissionLocalService, resourceName);
-		}
-		catch (NoSuchMethodException noSuchMethodException) {
-			resourcePermissions = resourcePermissionLocalService.getResourcePermissions(companyId, resourceName, ResourceConstants.SCOPE_INDIVIDUAL, String.valueOf(resourceId));
-		}
+		List<Resource> resources = _getResourceList(resourcePermissionSet);
```

Comment:
call this `getResources` and the same with the variable inside that method.

Reply:
I will also update some other method names, to maintain the consistency